src/Traits/DetectsExistingHttpClient.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace MyParcelCom\ApiSdk\Traits;
+
+use MyParcelCom\ApiSdk\Exceptions\MissingHttpClientAdapterException;
+use MyParcelCom\ApiSdk\Http\Contracts\HttpClient\ClientInterface;
+
+trait DetectsExistingHttpClient
+{
+    /**
+     * @return ClientInterface
+     * @throws MissingHttpClientAdapterException
+     */
+    private function detectExistingHttpClient()
+    {
+        if (class_exists('GuzzleHttp\Client')) {
+            // The BatchResults class was removed in Guzzle 6, so
+            // we assume we're at Guzzle 5
+            if (class_exists('GuzzleHttp\BatchResults')) {
+                // Return guzzle 5 adapter
+                if (!class_exists('Http\Adapter\Guzzle5\Client')) {
+                    throw new MissingHttpClientAdapterException('Missing cURL client package. Run `composer require php-http/guzzle5-adapter`.');
+                }
+
+                return new Http\Adapter\Guzzle5\Client();
+            }
+
+            // Otherwise return guzzle 6 adapter
+            if (!class_exists('Http\Adapter\Guzzle6\Client')) {
+                throw new MissingHttpClientAdapterException('Missing cURL client package. Run `composer require php-http/guzzle6-adapter`.');
+            }
+
+            return new Http\Adapter\Guzzle6\Client();
+        }
+
+        if (function_exists('curl_version')) {
+            // Return Curl client
+            if (!class_exists('Http\Client\Curl\Client')) {
+                throw new MissingHttpClientAdapterException('Missing cURL client package. Run `composer require php-http/curl-client`.');
+            }
+
+            return new Http\Client\Curl\Client();
+        }
+    }
+}
```

Comment:
This entire trait is actually unnecessary as you can get `php-http/discovery` to discover a client for you using the static method: `HttpClientDiscovery::find()`